### PR TITLE
Adds warning message when deterministic training loss stagnates too quickly in partial BNNs

### LINKED ISSUE #11: Add warning message when deterministic training loss stagnates too quickly in partial BNNs
I've encountered one dataset for which mini-batching in the deterministic neural network (DNN) training of a partial Bayesian neural network (PBNN) causes the DNN training loss to stagnate too quickly, resulting in significantly lower performance than non-Bayesian machine learning models. Increasing the batch size (or not doing any batching) alleviated the issue. It would be helpful if there was a check during DNN training that triggers a warning when the DNN training loss stagnates prematurely, offering a suggestion to increase the batch size as a possible solution.
dnn_loss

### PR DESCRIPTION
**Pull request by sarah-allec**
### Context
Certain deterministic NN hyperparameters may cause overfitting that manifests in the training loss initially decreasing very rapidly and stagnating early. A warning message to the user with suggested solutions would be helpful. Closes #11 

### Description
After the first epoch, the change in training loss is monitored and any time it drops more than 25% (see figure for justification of choice of 25%), a warning message is printed to the user:

`UserWarning: The deterministic training loss is decreasing rapidly - learning and accuracy may be improved by increasing the batch size, adjusting MAP sigma, or modifying the learning rate.`

dnn_loss

### Changes in the codebase
1. Added a function called `monitor_dnn_loss` in `neurobayes/utils/utils.py` that prints warning when loss has decreased by 25% at any epoch.
2. Added a call to `monitor_dnn_loss` in `flax_nets/deterministic_nn.py` in the training loop (`DeterministicNN.train()`).

### CODE REVIEW

**Review by ziatdinovmax — APPROVED** [review]
Looks good! There is a seemingly unrelated issue with python-3.10 tests failing, which I will need to figure out later, but this one is ready to be merged.
